skt/publisher.py

```diff
@@ -58,20 +58,16 @@ class CpPublisher(Publisher):
     TYPE = 'cp'
 
     def publish(self, source):
-        # FIXME Move expansion up the call stack, as this limits the class
-        # usefulness, because tilde is a valid path character.
-        shutil.copy(os.path.expanduser(source), self.destination)
+        shutil.copy(source, self.destination)
         return self.geturl(source)
 
 
 class ScpPublisher(Publisher):
     TYPE = 'scp'
 
     def publish(self, source):
-        # FIXME Move expansion up the call stack, as this limits the class
-        # usefulness, because tilde is a valid path character.
         subprocess.check_call(["scp",
-                               os.path.expanduser(source),
+                               source,
                                self.destination])
```

Comment:
Since the full call fits into one line now, can you merge those lines so it doesn't take up so much space?

Reply:
Yeah, I have intend to ask you about make absolute path for `tarpkg`, but it seems to me the `load_config` become longer. 
Ah I got what you mean. We should get absolute path for all configuration early. And then, we are not worry about path for later design